vm/systemSmartContracts/systemVMEEI.go

```diff
@@ -0,0 +1,34 @@
+package systemSmartContracts
+
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/vm"
+)
+
+type systemVMEEI struct {
+	vm.ContextHandler
+}
+
+// NewSystemVMEEI creates a new system vm eei context, mainly used for now in sovereign chain
+func NewSystemVMEEI(vmContext vm.ContextHandler) (*systemVMEEI, error) {
```

Comment:
This should be NewOneShardSystemVMEEI, the name is not good.

And should also have a rework on ProcessBuiltInFunction as well, without the sameShard IF in that case.